src/Cli/dotnet/Commands/Run/VirtualProjectBuildingCommand.cs

```diff
@@ -27,14 +30,69 @@ namespace Microsoft.DotNet.Cli.Commands.Run;
 /// </summary>
 internal sealed class VirtualProjectBuildingCommand
 {
+    /// <summary>
+    /// A file put into the artifacts directory when build starts.
+    /// It contains full path to the original source file to allow tracking down the input corresponding to the output.
+    /// It is also used to check whether the previous build has failed.
```

Comment:
This line seems misleading, it looks like the other string const is associated with detecting that the previous build succeeded or failed.

Possibly just one file was used here at first and then there was found to be a need to split into two files?

Reply:
The build-succeeded marker file could be used to detect failures if it was removed on failed builds. This might be more reliable though (what if file deletion failed as well). And it also provides more information (the input path for the artifacts).